tests/ui/pattern/deref-patterns/typeck.rs

```diff
@@ -0,0 +1,31 @@
+//@ check-pass
+#![feature(deref_patterns)]
+#![allow(incomplete_features)]
+
+use std::rc::Rc;
+
+fn main() {
+    let vec: Vec<u32> = Vec::new();
+    match vec {
+        box [..] => {}
```

Comment:
Why this `match vec { box [..] =>` compiles? How this is possible? You cannot move slice out of vector.

I think `match vec { box [..] =>` should fail to compile, and something like `match &vec { box [..] =>` should succeed

Reply:
Oops, I understood. This code compiles for the same reason this code compiles: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=76defa6bd202e1468eba8e65aaf2b7ab . I. e. because `match vec` doesn't always move out of `vec`. I'm sorry

Reply:
yep exactly, we match on the place, which doesn't require any moves